Update README and action description fields (#7)
SonarSource/sonarcloud-codecatalyst-action: README.md

## Usage

Project metadata, including the location of the sources to be analyzed, must be declared in the file `sonar-project.properties` in the base directory:

```properties
sonar.organization=<replace with your SonarCloud organization key>
sonar.projectKey=<replace with the key generated when setting up the project on SonarCloud>

# This is the name and version displayed in the SonarCloud UI.
#sonar.projectName=windows-msbuild-gh-actions-sq
#sonar.projectVersion=1.0
sonar.exclusions=venv/**

# relative paths to source directories. More details and properties are described
# in https://sonarcloud.io/documentation/project-administration/narrowing-the-focus/
sonar.sources=.

# Encoding of the source code. Default is default system encoding
#sonar.sourceEncoding=UTF-8
```
The workflow, usually declared in `.codecatalyst/workflows/build.yaml`, looks like:

```yaml
Name: SonarCloudAnalysis
SchemaVersion: "1.0"

# Optional - Set automatic triggers.
Triggers:
  - Type: Push
# Required - Define action configurations.
Actions:
  SonarCloudScanAction:
    # Identifies the action. Do not modify this value.

>     Identifier: sonar/sonarcloud-scan@v1.0.0

Author: For now, the name of the action is `sonarr66kw`, which is not ideal (see [the action here](https://codecatalyst.aws/spaces/Sonar/projects/SonarCloudScanAction/actions/sonar%2Fsonarr66kw/view)). 
I tried renaming it, but I don't have permission.
Once we publish the action we will not be able to change the name.

## Configuration parameters

- `SonarToken` – **Required** this is the token used to authenticate access to SonarCloud. You can generate a token on your [Security page in SonarCloud](https://sonarcloud.io/account/security/). You can set the `SONAR_TOKEN` environment variable in the "Secrets" settings page of your repository.
- `SonarOrganization` – **Optional** The key of the SonarCloud organization to which the project belongs. If not provided, the organization key will be read from the `sonar-project.properties` file.
- `SonarProjectKey` – **Optional** The key of the project on SonarCloud. If not provided, the project key will be read from the `sonar-project.properties` file.
- `SonarBranchName` – **Optional** The name of the branch being analyzed.
- `SonarProjectBaseDir` – **Optional** The base directory of the project to analyze. If not provided, the base directory will be the root of the repository.`
- `SonarAdditionalArgs` – **Optional** Additional arguments to pass to the SonarScanner CLI. For example, `-Dsonar.verbose=true` to enable verbose logging.

## Do not use this CodeCatalyst action if you are in the following situations

> * Your code is built with Maven: run 'org.sonarsource.scanner.maven:sonar' during the build

Reviewer: ```suggestion
* Your code is built with Maven: run 'org.sonarsource.scanner.maven:sonar-maven-plugin:3.11.0.3922:sonar' during the build (more info in the [SonarScanner for Maven](https://docs.sonarsource.com/sonarcloud/advanced-setup/ci-based-analysis/sonarscanner-for-maven/) documentation)
```

Author: > Your code is built with Maven: run 'org.sonarsource.scanner.maven:sonar-maven-plugin:3.11.0.3922:sonar' during the build (more info in the [SonarScanner for Maven](https://docs.sonarsource.com/sonarcloud/advanced-setup/ci-based-analysis/sonarscanner-for-maven/) documentation)

@agigleux (Can't reply directly on the comment) I am unsure about adding the specific version here, it would mean that every time the version is updated we would need to change it here as well.

Reviewer: We recommend in the SonarQube documentation to lock down the version of the Scanner for Maven to be in control and not get the latest changes and be surprised by the changes: https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/scanners/sonarscanner-for-maven/#other-settings

> * Your code is built with Gradle: use the SonarQube plugin for Gradle during the build

Reviewer: ```suggestion
* Your code is built with Gradle: use the SonarScanner for Gradle during the build (more info in the [SonarScanner for Gradle](https://docs.sonarsource.com/sonarcloud/advanced-setup/ci-based-analysis/sonarscanner-for-gradle/) documentation)
```

Author: What about removing what is between parenthesis and just having the link on `SonarScanner for Gradle` ?

Reviewer: OK. Let's do that.

> * You want to analyze a .NET solution: Follow our interactive tutorial for other CI's

Reviewer: ```suggestion
* You want to analyze a .NET solution: follow our interactive tutorial for other CI's
```

> * You want to analyze C and C++ code: rely on our [SonarCloud Scan for C and C++](https://github.com/marketplace/actions/sonarcloud-scan-for-c-and-c) and look at [our sample C and C++ project](https://github.com/sonarsource-cfamily-examples?q=gh-actions-sc&type=all&language=&sort=)

Author: I think this is still true actually. People can keep using our github action for C/C++ thanks to the `aws/github-actions-runner` action
